src/text/line.rs

```diff
@@ -474,12 +474,24 @@ impl<'a> Line<'a> {
             let new_span_width = span_width - offset;
             if x + new_span_width > result_width {
                 let span_end = (result_width - x + offset) as usize;
-                new_span.content = Cow::from(&span.content[offset as usize..span_end]);
+                new_span.content = Cow::from(
+                    span.content
+                        .as_ref()
+                        .chars()
+                        .skip(offset.into())
+                        .take(span_end - (offset as usize))
```

Comment:
.into() hides what is happening here. Being more explicit ends up in more readable code.
```suggestion
                        .skip(usize::from(offset))
                        .take(span_end - usize::from(offset))
```